internal/service/arcregionswitch/plan.go

```diff
@@ -0,0 +1,1606 @@
+// @Tags(identifierAttribute="arn")
+func newResourcePlan(context.Context) (resource.ResourceWithConfigure, error) {
+	r := &resourcePlan{}
+	return r, nil
+}
+
+type resourcePlan struct {
+	framework.ResourceWithConfigure
+	framework.WithImportByID
+}
+
+func (r *resourcePlan) ValidateModel(ctx context.Context, schema *fwschema.Schema) fwdiag.Diagnostics {
+	var diags fwdiag.Diagnostics
+	// Basic validation is handled by the schema validators
+	return diags
+}
+
+type resourcePlanModel struct {
+	Arn                          types.String                                         `tfsdk:"arn"`
+	ID                           types.String                                         `tfsdk:"id"`
+	Name                         types.String                                         `tfsdk:"name"`
+	ExecutionRole                types.String                                         `tfsdk:"execution_role"`
+	RecoveryApproach             fwtypes.StringEnum[awstypes.RecoveryApproach]        `tfsdk:"recovery_approach"`
+	Regions                      fwtypes.ListOfString                                 `tfsdk:"regions"`
+	Description                  types.String                                         `tfsdk:"description"`
+	PrimaryRegion                types.String                                         `tfsdk:"primary_region"`
+	RecoveryTimeObjectiveMinutes types.Int64                                          `tfsdk:"recovery_time_objective_minutes"`
+	AssociatedAlarms             fwtypes.SetNestedObjectValueOf[associatedAlarmModel] `tfsdk:"associated_alarms"`
+	Workflows                    fwtypes.ListNestedObjectValueOf[workflowModel]       `tfsdk:"workflow"`
+	Triggers                     fwtypes.ListNestedObjectValueOf[triggerModel]        `tfsdk:"triggers"`
+	Region                       types.String                                         `tfsdk:"region"`
+	Tags                         tftags.Map                                           `tfsdk:"tags"`
+	TagsAll                      tftags.Map                                           `tfsdk:"tags_all"`
+}
+
+// Custom expand to handle AssociatedAlarms set-to-map conversion
+// Expand converts the Terraform resource model to AWS API input.
+// Custom expansion is required because:
+// 1. Union Type Handling: ExecutionBlockConfiguration uses AWS SDK union types that AutoFlex cannot handle automatically
+// 2. Complex Nested Transformations: ScalingResources (list→map[string]map[string]) and RegionAndRoutingControls (list→map[string][]) require manual conversion
+// 3. Set-to-Map Conversions: AssociatedAlarms transforms from Terraform sets to AWS API maps with complex key generation
+// 4. Conditional Logic: Different execution block types require different field mappings and validations
+// AutoFlex works well for simple field mappings but cannot handle these complex structural transformations.
+func (m resourcePlanModel) Expand(ctx context.Context) (result any, diags fwdiag.Diagnostics) {
```

Comment:
The custom `Expand`/`Flatten` methods should only be defined on the data structures where AutoFlex does not work out of the box. E.g. The struct which maps to a union type. 

By defining the custom method on the root struct we're adding quite a bit of overhead by embedding many `flex.Expand` calls inside it rather than just calling once on the root object and deferring to custom handling only in the necessary branches of the structure.